metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java

```diff
@@ -142,6 +148,73 @@ public RecordTemplate getAspect(@Nonnull final Urn urn, @Nonnull final String as
         ebeanAspect -> toAspectRecord(urn, aspectName, ebeanAspect.getMetadata(), getEntityRegistry())).orElse(null);
   }
 
+  @Override
+  public EnvelopedAspect getLatestEnvelopedAspect(
+      @Nonnull String entityName,
+      @Nonnull Urn urn,
+      @Nonnull String aspectName) throws Exception {
+    return getLatestEnvelopedAspects(entityName, ImmutableSet.of(urn), ImmutableSet.of(aspectName)).getOrDefault(urn,
+        Collections.emptyList())
+        .stream()
+        .filter(envelopedAspect -> envelopedAspect.getName().equals(aspectName))
+        .findFirst()
+        .orElse(null);
+  }
+
+  @Override
+  public Map<Urn, List<EnvelopedAspect>> getLatestEnvelopedAspects(
+      @Nonnull String entityName,
+      @Nonnull Set<Urn> urns,
+      @Nonnull Set<String> aspectNames) throws Exception {
+
+    final Set<EbeanAspectV2.PrimaryKey> dbKeys = urns.stream()
+        .map(urn -> aspectNames.stream()
+            .map(aspectName -> new EbeanAspectV2.PrimaryKey(urn.toString(), aspectName, ASPECT_LATEST_VERSION))
+            .collect(Collectors.toList()))
+        .flatMap(List::stream)
+        .collect(Collectors.toSet());
+
+    final Map<EbeanAspectV2.PrimaryKey, EnvelopedAspect> envelopedAspectMap = getEnvelopedAspects(dbKeys);
+
+    // Group result by Urn
+    final Map<String, List<EnvelopedAspect>> urnToAspects = envelopedAspectMap.entrySet()
+        .stream()
+        .collect(Collectors.groupingBy(entry -> entry.getKey().getUrn(),
+            Collectors.mapping(Map.Entry::getValue, Collectors.toList())));
+
+    // For each input urn, get the aspects corresponding to the urn. If empty, return the key aspect
+    final Map<Urn, List<EnvelopedAspect>> result = new HashMap<>();
+    for (Urn urn : urns) {
+      List<EnvelopedAspect> aspects = urnToAspects.getOrDefault(urn.toString(), Collections.emptyList());
+      EnvelopedAspect keyAspect = getKeyEnvelopedAspect(urn);
+      // Add key aspect if it does not exist in the returned aspects
+      if (aspects.isEmpty() || !aspectNames.contains(keyAspect.getName())) {
+        result.put(urn, ImmutableList.<EnvelopedAspect>builder().addAll(aspects).add(keyAspect).build());
```

Comment:
Actually unsure that we should do this. I think we'd previously had it but it seems kinda strange to inject an aspect that was not explicitly asked for. WDYT?

Reply:
Agreed. I wasn't sure if this would break things.